mocked_servers/grpc_trace/main.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"log"
+	"net"
+	"net/http"
+	"time"
+
+	pb "go.opentelemetry.io/proto/otlp/collector/trace/v1"
+	"google.golang.org/grpc"
+)
+
+const (
+	HealthCheckMessage = "healthcheck"
+	SuccessMessage     = "success"
+)
+
+type server struct {
+	data *string
+	pb.UnimplementedTraceServiceServer
+}
+
+type checkDataHandler struct {
+	data *string
+}
+
+func healthCheck(w http.ResponseWriter, _ *http.Request) {
+	if _, err := w.Write([]byte(HealthCheckMessage)); err != nil {
+		log.Printf("Unable to write response: %v", err)
+	}
+}
+
+func (h *checkDataHandler) ServeHTTP(w http.ResponseWriter, _ *http.Request) {
+	if _, err := w.Write([]byte(*h.data)); err != nil {
+		log.Printf("Unable to write response: %v", err)
+	}
+}
+
+// Export Implements the RPC method.
+func (s *server) Export(_ context.Context, _ *pb.ExportTraceServiceRequest) (*pb.ExportTraceServiceResponse, error) {
```

Comment:
Does this require a completely separate implementation?  This looks almost identical to the metrics implementation.

Reply:
It's identical just uses different protos.